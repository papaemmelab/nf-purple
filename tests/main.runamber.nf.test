nextflow_process {

    name "Test Process runAmber"
    script "main.nf"
    process "runAmber"

    test("Should run Amber without failures") {

        when {
            params {
                tumor = "TEST_TUMOR"
                normal = "TEST_NORMAL"
                tumorBam = "${projectDir}/tests/data/tumor.bam"
                normalBam = "${projectDir}/tests/data/normal.bam"
                outdir = "${projectDir}/tests/outdir"
            }
            process {
                """
                input[0] = Channel.value(params.tumor)
                input[1] = Channel.value(params.normal)
                input[2] = Channel.fromPath(params.tumorBam)
                input[3] = Channel.fromPath(params.normalBam)
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
            assert process.trace.tasks().size() == 1
            
            // check expected files
            with(process.out) {
                assert amber_baf_pcf.size() == 1
                assert amber_baf_tsv.size() == 1
                assert amber_qc.size() == 1
            }
        }

    }

}
