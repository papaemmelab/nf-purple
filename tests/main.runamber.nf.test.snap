{
    "Should run Amber without failures": {
        "content": [
            {
                "0": [
                    "TEST_TUMOR.amber.baf.tsv.gz:md5,582a86aee1eb90035f3a5d22cf6cdbd9"
                ],
                "1": [
                    "TEST_TUMOR.amber.baf.pcf:md5,c8d8a68055028ceea067b3c53d55aae4"
                ],
                "2": [
                    "TEST_TUMOR.amber.qc:md5,9547e68ba672e08d6bd49e1e6f654ee8"
                ],
                "3": [
                    "TEST_TUMOR.amber.contamination.vcf.gz:md5,941abd5cb8324ae2eabe82d9d0ebb6ee"
                ],
                "4": [
                    "TEST_NORMAL.amber.snp.vcf.gz:md5,baa474a32f1fd4dd54989d490470924c"
                ],
                "5": [
                    "TEST_NORMAL.amber.homozygousregion.tsv:md5,1d396d14d89860c41106934d2b5e16ac"
                ],
                "amber_baf_pcf": [
                    "TEST_TUMOR.amber.baf.pcf:md5,c8d8a68055028ceea067b3c53d55aae4"
                ],
                "amber_baf_tsv": [
                    "TEST_TUMOR.amber.baf.tsv.gz:md5,582a86aee1eb90035f3a5d22cf6cdbd9"
                ],
                "amber_contamination_vcf": [
                    "TEST_TUMOR.amber.contamination.vcf.gz:md5,941abd5cb8324ae2eabe82d9d0ebb6ee"
                ],
                "amber_normal_homozygousregion_tsv": [
                    "TEST_NORMAL.amber.homozygousregion.tsv:md5,1d396d14d89860c41106934d2b5e16ac"
                ],
                "amber_normal_snp_vcf": [
                    "TEST_NORMAL.amber.snp.vcf.gz:md5,baa474a32f1fd4dd54989d490470924c"
                ],
                "amber_qc": [
                    "TEST_TUMOR.amber.qc:md5,9547e68ba672e08d6bd49e1e6f654ee8"
                ]
            }
        ],
        "timestamp": "2024-01-23T17:24:43.748894"
    }
}